nck/readers/mytarget_reader.py

```diff
@@ -0,0 +1,252 @@
+# GNU Lesser General Public License v3.0 only
+# Copyright (C) 2020 Artefact
+# licence-information@artefact.com
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+import click
+import nck.helpers.api_client_helper as api_client_helper
+from nck.clients.api_client import ApiClient
+from nck.commands.command import processor
+from nck.readers.reader import Reader
+from nck.streams.json_stream import JSONStream
+from nck.utils.args import extract_args
+import requests
+from typing import Dict
+from nck.utils.secret_retriever import get_secret
+from botocore.config import Config
+import logging
+import json
+import boto3
+import itertools
+
+
+@click.command(name="read_mytarget")
+@click.option("--mytarget-client-id", required=True)
+@click.option("--mytarget-client-secret", required=True)
+@click.option("--mytarget-mail", required=True)
+@click.option("--mytarget-agency", required=True)
+@processor("mytarget-client-id", "mytarget-client-secret")
+def mytarget(**kwargs):
+    return MyTargetReader(**extract_args("mytarget_", kwargs))
+
+
+secret_keys = json.loads(get_secret(
+    'arn:aws:secretsmanager:eu-west-3:127773427021:secret:access-key-sanofi-ANrWKh',
+    'eu-west-3')
+)
+
+
+s3_resource = boto3.resource(
+    's3',
+    aws_access_key_id=secret_keys.get('ACCESS_KEY'),
+    aws_secret_access_key=secret_keys.get('SECRET_KEY'),
+    region_name='eu-west-3',
+    config=Config(
+        retries={
+            'max_attempts': 10
+        }
+    )
+)
+
+
+class MyTargetReader(Reader):
+
+    def __init__(
+        self,
+        client_id,
+        client_secret,
+        mail,
+        agency,
+        **kwargs
+    ):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.mail = mail
+        self.agency = agency
+        self.access_token = None
+        self.agency_client_token = None
+
+    def create_agency_token_request(self, agency_client):
+        url = 'https://target.my.com/api/v2/oauth2/token.json'
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Host': 'target.my.com'
+        }
```

Comment:
This can be considered as config. I know in other readers, there are similar hardcoded URLs and headers in class methods but we should avoid that as much as possible. At least, here, we could set the url as a class constant if we don't want to create a new config file.

Reply:
Ok so keep the same functions but just get the url from a config file ? If we use config files it may be more interesting add all the parameters that vary from one function to another such as 'params' or 'payload' dicts right ?

Reply:
Yes absolutely. Url + payloads/headers/... Everything that is static. Even if a dict is populated at runtime we can use template strings.

Reply:
Going to take care of this right away.